tutor/commands/plugins.py

```diff
@@ -135,15 +136,13 @@ def list_command(show_enabled_only: bool) -> None:
 @click.argument("plugin_names", metavar="plugin", nargs=-1, type=PluginName())
 @click.pass_obj
 def enable(context: Context, plugin_names: list[str]) -> None:
-    config = tutor_config.load_minimal(context.root)
+    config = tutor_config.load_full(context.root)
```

Comment:
Aaaaand you just broke your config.yml! If you load the full configuration, then your config.yml will also store all configuration settings, including the default ones. This means that they will not be updated whenever we modify the default values in config/defaults.yml.

Only the minimal configuration must be used to save the config.yml yml file -- whereas the full configuration is needed to update the environment.

This is one of the trickiest parts of the tutor code base, sorry you got bit :-/ I'm not sure what's the best way to bypass this issue. Maybe call the `save` command, which does things properly?

Reply:
> Aaaaand you just broke your config.yml! If you load the full configuration

😄 Yeah, you're right.

> sorry you got bit :-/ 

No worries 😄  I've always preferred a good code mystery over easy-mode anyway!

> Maybe call the save command, which does things properly?

I faced something weird. for the `enable` command, I managed to invoke `save` using this:

```
@click.command(help="Enable a plugin")
@click.argument("plugin_names", metavar="plugin", nargs=-1, type=PluginName())
@click.pass_context
def enable(context: click.Context, plugin_names: list[str]) -> None:
    if not plugin_names:
        fmt.echo_error("No plugin names provided.")
        return
    config = tutor_config.load_minimal(context.obj.root)
    for plugin in plugin_names:
        plugins.load(plugin)
        fmt.echo_info(f"Plugin {plugin} enabled")
    tutor_config.save_enabled_plugins(config)
    tutor_config.save_config_file(context.obj.root, config)
    context.invoke(config_save_command, env_only=True)
```
But for the `disable` command, this approach fails. it tries to render parts of the removed plugin, even though it's not a part of `config.yml`. for example, after I disabled the `mfe` plugin: 

```
Error rendering patch 'caddyfile':
{{ MFE_HOST }}{$default_site_port} {
    respond / 204
    request_body {
        max_size 2MB
    }
    import proxy "mfe:8002"
}

Error rendering template apps/caddy/Caddyfile
Error: Missing configuration value: 'MFE_HOST' is undefined
```

Any thoughts?

Reply:
Can you please add a unit test that reproduces the issue? Otherwise it's difficult for us to troubleshoot.